platform/linux-dpdk/include/odp/api/plat/version_types.h

```diff
@@ -0,0 +1 @@
+../../../../../linux-generic/include/odp/api/plat/version_types.h
```

Comment:
This is my understanding of the **odp-dpdk-import** proposal (best viewed on GitHub instead of email):

- **master**
- **odp-dpdk-import**
  - Branch off: **master**
    - Lifetime: short-lived (4-8w)
    - Sync strategy:
      - Sync with master often
        - Dependent on **master** sync'ing from **api-next**
      - Fold back to **master**
  - Activity 1:
    - Copy over odp-dpdk.git /platform/linux-dpdk/
    - Deprecate odp-dpdk.git
      - Celebrate achieving 1 unified repository (odp.git)
    - Rename to ODP Cloud (e.g. /platform/odp-cloud/)
    - Address technical debt (more than 1 platform in odp.git):
      - Move shared code to common location
      - Further improvements to build system
- **cloud-dev**
  - Branch off: **odp-dpdk-import**
    - Lifetime: long-lived
    - Sync strategy:
      - Sync with **odp-dpdk-import** often
      - Rebase on to **master** when **odp-dpdk-import** folds
      - When to fold back to **master** ?
  - Activity 2:
    - Directory structure (Bogdan Honnappa doc)
    - Modular Framework
      - Loadable platforms (subsystem providers)
      - DPDK "platform" (source code independence)
    - Virt/Native drivers
    - Buffer improvements

We seem to agree that both **odp-dpdk-import** and **cloud-dev** work is needed, and that both can run in parallel. Now, it is just a question of: _where?_

A branch creates a vacuum for an engineer. It allows her to proceed with work while, essentially, frozen in time. Since the work needs to be contributed back to **master**, it is a good idea to 'phone home' occasionally--this is achieved via branch syncs and/or simply short-lived branches. Conflicts cannot be avoided. If there is some issue with software, it needs to be handled whether everyone works off master or whether everyone uses branches that eventually fold into master.

Since **odp-dpdk-import** and **cloud-dev** work will be highly dependent on each other, it is best for both to share a branch. This prevents (potentially critical) issues from lingering around until sync/merge time _and_ eliminates the overhead of having to execute branch syncs!

We could even make that branch **master** instead of using a separate branch in the first place.  This means that all **odp-dpdk-import** and **cloud-dev** work is in sync with **master**. The majority of the work will be contained under /platform/odp-cloud/ _separately_ from /platform/linux-generic/ so it will be good to use the same **master** branch--the only exception being the "Directory structure" proposal.

As for _how?_...

The code in odp-dpdk.git repo that is being pulled into odp.git repo via **odp-dpdk-import** has already undergone review to get into the odp-dpdk.git repo in the first place. I think I share the same paranoi as @Bill-Fischofer-Linaro and @lumag about migrating over such a large chunk of code, and that is why I reviewed it. I found a couple minor things, but those relate to the **api-next** branch which points towards the struggles of maintaining long-lived branches. Other than that, I am very happy to have this code contributed so that we can execute on the above plans. Preferably all in the same branch.

What does this look like?

1. Eliminate branches (everything on **master**)
2. Get /platform/odp-cloud/ in
3. Run "technical debt" and "odp-cloud" in parallel

- **master**
  - Activity 1:
    - Copy over odp-dpdk.git /platform/linux-dpdk/
    - Deprecate odp-dpdk.git
      - Celebrate achieving 1 unified repository (odp.git)
    - Rename to ODP Cloud (e.g. /platform/odp-cloud/)
  - Activity 2a (parallel)
    - Address technical debt (more than 1 platform in odp.git):
      - Move shared code to common location
      - Further improvements to build, test, and docs
  - Activity 2b (parallel)
    - Directory structure (Bogdan Honnappa doc)
    - Modular Framework
      - Loadable platforms (subsystem providers)
      - DPDK "platform" (source code independence)
    - Virt/Native drivers
    - Buffer improvements

Notice zero time spent on branch management, and zero risk due to deferring issues until branch sync/merge.